volatility3/framework/plugins/windows/modules.py

```diff
@@ -165,13 +165,33 @@ def get_session_layers(
 
                 # create the session space object in the process' own layer.
                 # not all processes have a valid session pointer.
-                session_space = context.object(
-                    symbol_table + constants.BANG + "_MM_SESSION_SPACE",
-                    layer_name=layer_name,
-                    offset=proc.Session,
-                )
-
-                if session_space.SessionId in seen_ids:
+                try:
+                    session_space = context.object(
+                        symbol_table + constants.BANG + "_MM_SESSION_SPACE",
+                        layer_name=layer_name,
+                        offset=proc.Session,
+                    )
+                    session_id = session_space.SessionId
+
+                except exceptions.SymbolError:
+                    # In Windows 11 24H2, the _MM_SESSION_SPACE type was
+                    # replaced with _PSP_SESSION_SPACE, and the kernel PDB
+                    # doesn't contain information about its members (otherwise,
+                    # we would just fall back to the new type). However, it
+                    # appears to be, for our purposes, functionally identical
+                    # to the _MM_SESSION_SPACE. Because _MM_SESSION_SPACE
+                    # stores its session ID at offset 8 as an unsigned long, we
+                    # create an unsigned long at that offset and use that
+                    # instead.
+                    session_id = int(
```

Comment:
This shouldn't need coercing to an int, is there a reason for doing this?  Since the base types should be based off of normal ints, comparison and equality (for example, for set membership) *should* all work normally (but please check, if it doesn't it might be better to fix that for the whole base type rather than doing casts).

Reply:
Fixed, added a type hint to the method signature as well.